src/topology/sheaves/sheaf_condition/sites.lean

```diff
@@ -473,3 +473,49 @@ begin
 end
 
 end Top.presheaf
+
+namespace Top.opens
+
+open category_theory topological_space
+
+variables {X : Top} {ι : Type*} [category ι]
+
+lemma cover_dense_iff_basis (B : ι ⥤ opens X) :
+  cover_dense (opens.grothendieck_topology X) B ↔ opens.is_basis (set.range B.obj) :=
+begin
+  rw opens.is_basis_iff_nbhd,
+  split, intros hd U x hx, rcases hd.1 U x hx with ⟨V,f,⟨i,f₁,f₂,hc⟩,hV⟩,
+  exact ⟨B.obj i, ⟨i,rfl⟩, f₁.le hV, f₂.le⟩,
+  intro hb, split, intros U x hx, rcases hb hx with ⟨_,⟨i,rfl⟩,hx,hi⟩,
+  exact ⟨B.obj i, ⟨⟨hi⟩⟩, ⟨i, 𝟙 _, ⟨⟨hi⟩⟩, by rw category.id_comp⟩, hx⟩,
+end
+
+lemma cover_dense_induced_functor {B : ι → opens X} (h : opens.is_basis (set.range B)) :
+  cover_dense (opens.grothendieck_topology X) (induced_functor B) :=
+(cover_dense_iff_basis _).2 h
+
+end Top.opens
+
+namespace Top.sheaf
+
+open category_theory topological_space Top opposite
+
+variables {C : Type u} [category.{v} C] [limits.has_products C]
+variables {X : Top.{v}} {ι : Type v} {B : ι → opens X}
+variables (F : presheaf C X) (F' : sheaf C X) (h : opens.is_basis (set.range B))
+
+def restrict_hom_equiv_hom :
+  ((induced_functor B).op ⋙ F ⟶ (induced_functor B).op ⋙ F'.1) ≃ (F ⟶ F'.1) :=
+@cover_dense.restrict_hom_equiv_hom _ _ _ _ _ _ _ _
+  (@opens.cover_dense_induced_functor X ι (induced_category.category B) B h)
+  _ F ((presheaf.Sheaf_spaces_to_sheaf_sites C X).obj F')
+
+lemma extend_hom_app {α i} :
```

Comment:
Would this benefit from being a simp lemma?
Also, some (implicit?) style guide seems to suggest avoiding implicit type arguments whenever possible.

Reply:
The type of `α` is too long in my opinion. I was tempted to write `∀ {α i}` but found that this also works.

Reply:
The thing is that I cannot find out the type of `α` at first glance without spending time finding `α` in the lemma statement, and then finding the corresponding definition of `restrict_hom_equiv_hom`, and then looking at the type of the definition to find out which part is my `α` referring to.

In my opinion, a type that fits in a line is probably not too long to be explicitly written out.

Reply:
I find `simp` and `dsimp` are generally very slow in this part of mathlib, and I didn't use them in the whole main module Gamma_Spec_adjunction except for a `simp only`. Thus I hesitate to introduce any `simp` lemmas or `instance`s. Others are free to add them if they make sure mathlib still builds in time, I guess.

Reply:
Not so sure about simp, but new instances does not tend to slow down stuff in my experience.

Reply:
My experience is that the more simp lemmas, the faster things will go, since Lean will rewrite things into a better form without unfolding definitions and going too deep. But I guess some more experienced Lean user could give some thoughts on this.

Reply:
> My experience is that the more simp lemmas, the faster things will go, since Lean will rewrite things into a better form without unfolding definitions and going too deep.

You're saying more simp lemmas could speed up the `simp` tactic. Good to know! Maybe we have too few of them right now then.

> The thing is that I cannot find out the type of α at first glance 

But in the documentation the full type will be there, right?

Reply:
My rationale is that some people will use the source code itself as the documentation.
At least in my opinion, the readability of the source code is something that is worth pursuing.
But in the end it depends on you (or probably the moderators) whether it is worth the change in this specific case.

Reply:
Full types have been supplied.